src/main/scala/it/pagopa/interop/partyprocess/api/impl/ProcessApiServiceImpl.scala

```diff
@@ -1287,12 +1287,17 @@ class ProcessApiServiceImpl(
       )
 
       institutionInternalId = institutions.headOption.map(_.id.toString).getOrElse("")
+      institutionExternalId = institutions.headOption.map(_.externalId.toString).getOrElse("")
+
+      institution <- institutions.headOption.toFuture(
+        InstitutionNotFound(Option(institutionInternalId), Option(institutionExternalId))
+      )
 
       productId    = managerRelationships.head.product.id
       managerUsers = managerRegistryUsers.map(u => createUser(managerRelationships, institutionInternalId, u))
-
-      manager             = managerRegistryUsers.head
-      managerRelationship = managerRelationships.filter(_.from == manager.id).head
+      managerUser         <- managerUsers.headOption.toFuture(ManagerFoundError)
+      manager             <- managerRegistryUsers.headOption.toFuture(ManagerFoundError)
+      managerRelationship <- managerRelationships.filter(_.from == manager.id).headOption.toFuture(ManagerFoundError)
```

Comment:
just FYI, this is equivalent to
```suggestion
      managerRelationship <- managerRelationships.find(_.from == manager.id).toFuture(ManagerFoundError)
```